File: Python/Permutations.py
def permute(nums):
    output = []

    if len(nums)==1:
        return [nums.copy()]
    
    for i in range(len(nums)):
        n = nums.pop()
        permutations = permute(nums)

        for permutation in permutations:
            permutation.append(n)

        output.extend(permutations)
        nums.append(n)

    return output

File: Javascript/Permutations.js
var permute = function(nums){
    const output = []

    if(nums.length === 1){
        return [[...nums]]
    }

    for(let i=0;i<nums.length;i++){
        const n = nums.shift()
        const permutations = permute(nums)

        permutations.forEach((permutation) => permutation.push(n))

        output.push(...permutations)
        nums.push(n)
    }

    return output
}